base/abstractarraymath.jl

```diff
@@ -166,15 +166,7 @@ end
 
 ## Permute array dims ##
 
-function permutedims(B::AbstractArray, perm)
-    dimsB = size(B)
-    ndimsB = length(dimsB)
-    (ndimsB == length(perm) && isperm(perm)) || throw(ArgumentError("no valid permutation of dimensions"))
-    dimsP = ntuple(i->dimsB[perm[i]], ndimsB)::typeof(dimsB)
-    P = similar(B, dimsP)
-    permutedims!(P, B, perm)
-end
-
+permutedims(B::AbstractArray, perm) = permutedims(convert(Array, B), perm)
 
```

Comment:
Note in particular that this is making a copy of the input, creating the output, and then throwing away the copy of the input. Far from ideal, but better than throwing an error.

Writing the general version for arbitrary dimensions isn't entirely trivial, but neither is it likely to be really difficult. Probably the best approach would be to create a `PermutedDimsArray` wrapper-type that swaps the indexes supplied to `getindex/setindex!` before passing them on to the parent array. If we don't export it (using it basically only to support this particular function), then maybe we wouldn't need to address all the optimized-iteration concerns just yet.